### PR TITLE
Update sentry to fix error with the Privacy files

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
On https://github.com/mattermost/mattermost-mobile/pull/7891 I added the privacy files, but I made the big mistake of not running the app after setting this up. The result is that the `sentry` package was also declaring this privacy file and making conflicts with our own.

This was solved on later versions of sentry, so I updated the dependency to the latest, as https://github.com/mattermost/mattermost-mobile/pull/7863 was going to do either way.

The idea to push this PR separately is because right now it is not possible to create a build because of this failure.

#### Ticket Link
None

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,24 +1,6 @@
...
-      },
-      {
-        "package": "Sentry",
-        "repositoryURL": "https://github.com/getsentry/sentry-cocoa.git",
-        "state": {
-          "branch": "8.15.2",
-          "revision": "c60f232c96264c3bea75bc0b81e6126b2f7475ee",
-          "version": null
-        }
-      },
```
Comment (enahum) [inline ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved:21]: this does not seem to be on purpose

File: ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -1,24 +1,6 @@
...
-          "branch": null,
-          "revision": "382972f1963580eeabafd88ad012e66576b4d213",
-          "version": "1.4.1"
-        }
-      },
-      {
-        "package": "Sentry",
-        "repositoryURL": "https://github.com/getsentry/sentry-cocoa.git",
-        "state": {
-          "branch": "8.15.2",
```
Comment (enahum) [inline ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved:17]: you should bump this one to match the one with the current sentry RN package which is 8.21.0

**Review by enahum — CHANGES_REQUESTED** [review]

File: ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (larkox) [reply ios/Mattermost.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Updating the version touched the right project file places for this to never happen again, hopefully...

**Review by enahum — APPROVED** [review]

**Review by mvitale1989 — APPROVED** [review]
The build pipelines are working again now, thanks Daniel!

**Review by cpoile — APPROVED** [review]
Thank you @larkox!

**Pull request merged**